test/unit/lib/runner/writers/action-json-writer-tests.js

```diff
@@ -0,0 +1,798 @@
+              'status': 'pass',
+              'steps': {
+                'effects': {
+                  'error': null,
+                  'status': 'pass',
+                },
+                'perform': {
+                  'error': null,
+                  'status': 'pass',
+                },
+                'precondition': {
+                  'error': null,
+                  'status': 'pass',
+                },
+              },
+              'time': [
+                1,
+                201697760,
+              ],
+            },
+            {
+              'averageTime': NaN,
+              'count': 1,
+              'failures': [],
+            },
+          ],
+        ]);
+      });
+    });
+  });
+
+  describe('_getActionTestData', function() {
+    let actionJsonWriter;
+
+    beforeEach(function() {
+      mockery.enable({useCleanCache: true});
+      mockery.registerAllowable('../../../../../lib/runner/writers/action-json-writer.js');
+      mockery.registerMock('fs', {});
+      mockery.registerMock('path', {});
+      mockery.registerMock('../../util/config-handler.js', {});
+
+      actionJsonWriter = require('../../../../../lib/runner/writers/action-json-writer.js');
+    });
+
+    afterEach(function() {
+      mockery.resetCache();
+      mockery.deregisterAll();
+      mockery.disable();
+    });
+
+    describe('if there is no test data in actionJsonWriter._testData[actionName]', function() {
+      it('should _testData[actionName] to an object with the data', function() {
+        let actionName = 'action';
+        actionJsonWriter._testData[actionName] = false;
+
+        let testData = actionJsonWriter._getActionTestData(actionName);
+
+        expect(testData).to.deep.equal({
+          count: 0,
+          failures: [],
+          averageTime: 0,
+          status: 'pass',
+        });
+      });
+    });
+
+    describe('if there is test data in actionJsonWriter._testData[actionName]', function() {
+      it('should add  it', function() {
+        let actionName = 'action';
+        actionJsonWriter._testData[actionName] = true;
+
+        let testData = actionJsonWriter._getActionTestData(actionName);
+
+        expect(testData).to.deep.equal(true);
+      });
+    });
+  });
+
+  describe('_createNotes', function() {
+    let actionJsonWriter;
+
+    beforeEach(function() {
+      mockery.enable({useCleanCache: true});
+      mockery.registerAllowable('../../../../../lib/runner/writers/action-json-writer.js');
+      mockery.registerMock('fs', {});
+      mockery.registerMock('path', {});
+      mockery.registerMock('../../util/config-handler.js', {});
+
+      actionJsonWriter = require('../../../../../lib/runner/writers/action-json-writer.js');
+    });
+
+    afterEach(function() {
+      mockery.resetCache();
+      mockery.deregisterAll();
+      mockery.disable();
+    });
+
+    describe('for each failure within test.failures', function() {
+      it('should return report notes', function() {
```

Comment:
you need a test that has no failures to make sure the default reportNotes are made, and one with errors to make sure the reportNotes is modified as you expect